.github/workflows/test.yaml

```diff
@@ -1,39 +1,215 @@
+---
 name: Test
 on:
   push:
     branches:
       - master
-      - 'release/**'
-  pull_request:
+      - main
+      - release/**
+  pull_request: null
+  workflow_dispatch:
+
 jobs:
   golangci-lint:
     runs-on: ubuntu-22.04
     timeout-minutes: 20
     steps:
-      - uses: actions/checkout@v3.0.2
+      - uses: actions/checkout@v4.1.1
         with:
           fetch-depth: 1
       - uses: actions/setup-go@v3
         with:
           go-version: 1.19.x
       - run: sudo apt-get update && sudo apt-get install -y libseccomp-dev
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.2.0
+        uses: golangci/golangci-lint-action@v3.7.0
         with:
-          version: v1.49.0
+          version: v1.55.2
           args: --verbose
-  ubuntu-2204:
-    name: "Ubuntu 22.04"
-    # nested virtualization is only available on macOS hosts
-    runs-on: macos-12
-    timeout-minutes: 30
+
+  create-lxc-image:
+    name: create-lxc-image
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4.1.1
+
+      - uses: actions/cache/restore@v3
+        id: cache-restore
+        with:
+         key: lxc-image-base-${{ hashFiles('go.sum', 'test/init_test.sh') }}
+         path: /tmp/test-image.tar.zst
+         lookup-only: true
+
+      - name: setup lxd (v5.19)
+        id: s1
+        if: steps.cache-restore.outputs.cache-hit != 'true'
+        run: ./setup_lxd.sh
```

Comment:
Can we move these scripts into a subdirectory?

Reply:
fixed in https://github.com/rootless-containers/bypass4netns/pull/39/commits/4b874fee803256d87c99a4765fd297743f3f5d5f